### PR TITLE
Jali: Fix bugs in CMake section

### PR DESCRIPTION
**Pull request by raovgarimella**
Fix mistakes in CMake commands block

### CODE REVIEW

File: var/spack/repos/builtin/packages/jali/package.py [hunk 1]
```diff
@@ -44,15 +44,15 @@ class Jali(CMakePackage):
 
     def cmake_args(self):
         options = []
-        if '+with_mstk' in self.spec:
+        if '+mstk' in self.spec:
             options.append('-DENABLE_MSTK_Mesh=ON')
         else:
             options.append('-DENABLE_MSTK_Mesh=OFF')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/jali/package.py]: Writing this block as
```python
options.append(self.define_from_variant('ENABLE_MSTK_Mesh', 'mstk'))
```
checks the variant spelling so this won't won't happen again.

File: var/spack/repos/builtin/packages/jali/package.py [hunk 2]
```diff
@@ -44,15 +44,15 @@ class Jali(CMakePackage):
...
 
         # Unit test variant
         if self.run_tests:
-            options.append('-DENABLE_Tests=ON')
+            options.append('-DENABLE_TESTS=ON')
         else:
-            options.append('-DENABLE_Tests=OFF')
+            options.append('-DENABLE_TESTS=OFF')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/jali/package.py]: Likewise, this block can be written as 
```python
options.append(self.define('ENABLE_TESTS', self.run_tests))
```

**Review by sethrj — CHANGES_REQUESTED** [review]
CMakePackage helper functions improve robustness and reduce code size and duplication.

File: var/spack/repos/builtin/packages/jali/package.py (hunk 1 above)
Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: Thanks. Updated.

File: var/spack/repos/builtin/packages/jali/package.py (hunk 2 above)
Comment (raovgarimella) [reply var/spack/repos/builtin/packages/jali/package.py]: Yes, Thank you for the hint. Fixed.

**Review by sethrj — APPROVED** [review]

**Pull request merged**